### PR TITLE
Add tests for spritesheet and file animations

### PR DESCRIPTION
**Pull request by gunchleoc**
To ensure that all supported animation types and helper functions will work

### CODE REVIEW

File: test/maps/plain.wmf/scripting/tribes/spritesheet_animation/init.lua [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+   },
+
+   indicate_workarea_overlaps = {
+      barbarians_spritesheet_animation = false
+   },
+
+   spritesheets = {
+      idle = {
+         directory = dirname,
+         basename = "idle",
```
Comment (Noordfrees) [inline test/maps/plain.wmf/scripting/tribes/spritesheet_animation/init.lua]: Please convert all the animation definitions to the new format: Get rid of `pictures`, `basename` and `directory`, add `animation_directory`. Same for the worker.

**Comment by gunchleoc** [comment]
At the moment, the test suite doesn't survive a stable_save with these custom units. I'll wait for the tribes filtering branch before fixing this, because the code for custom units will change there.

**Comment by gunchleoc** [comment]
tribes-filtering fixed this branch, so it is ready for review again.

**Comment by gunchleoc** [comment]
The code review has been addressed, so I'm merging.

**Pull request merged**